metagpt/roles/kaggle_manager.py

```diff
@@ -0,0 +1,153 @@
+import json
+import os
+import subprocess
+
+import fire
+import pandas as pd
+
+from metagpt.actions import Action, UserRequirement
+from metagpt.actions.ml_da_action import SummarizeAnalysis
+from metagpt.config import CONFIG
+from metagpt.logs import logger
+from metagpt.roles import Role
+from metagpt.schema import Message
+from metagpt.utils.common import CodeParser
+
+os.environ["KAGGLE_USERNAME"] = CONFIG.kaggle_username
+os.environ["KAGGLE_KEY"] = CONFIG.kaggle_key
+
+
+def run_command(cmd):
+    print(cmd)
+    output = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+    if output.returncode != 0:
+        print("Error output:", output.stderr)
+        exit()
+    else:
+        print(output.stdout)
+    return output.stdout
+
+
+class DownloadData(Action):
+    async def run(self, competition, data_desc="") -> str:
+        data_path = CONFIG.workspace_path / competition
+
+        output = run_command(f"kaggle competitions list --search {competition}")
+        assert output != "No competitions found", "You must provide the correct competition name"
+
+        run_command(f"kaggle competitions download {competition} --path {WORKSPACE_ROOT}")
+
+        if not os.path.exists(data_path):
+            # if True:
+            # run_command(f"rm -r {data_path / '*'}")
+            run_command(f"unzip -o {CONFIG.workspace_path / '*.zip'} -d {data_path}")  # FIXME: not safe
+
+        file_list = run_command(f"ls {data_path}")
+
+        rsp = f"""
+        Location:
+        Data downloaded at {data_path} folder, including {file_list}
+        Data Description:
+        {data_desc}
+        """
+        return rsp
+
+
+class SubmitResult(Action):
+    PROMPT_TEMPLATE: str = """
+    # Summary
+    __summary__
+    # Your task
+    Extract the file path for test set prediction from the summary above, output a json following the format:
+    ```json
+    {"file_path": str = "the file path, for example, /path/to/the/prediction/file/xxx.csv, /path/to/the/prediction/file/xxx.xlsx"}
+    ```
```

Comment:
use ActionNode and dont write __summary__
to solve {} error in f-string, use {{}} instead

Reply:
experimental code, deleted